back-end/VivaLaDama/VivaLaDama/Models/Chessboard.cs

```diff
@@ -64,42 +75,45 @@ public void FindPossibleDestination(Move move, out Coordinate dest1, out Coordin
 
             dest1 = dest2 = null;
 
-            if(this.IsMovementValid(from, to))
+            if (this.IsMovementValid(from, to))
```

Comment:
Negli if successivi ho visto che verifichi che il pawn sia upgraded, perchè verificarlo in tutte le condizioni? Si potrebbe mettere all'interno dell'if in testa in modo da verificarlo una volta per tutte ed evitare duplicazione?